### DISCUSSION TITLE
Solution for Customer Profile Integration

### DISCUSSION THREAD

**Discussion by texiontech** [solidusio/solidus#4864] [Troubleshooting]
Hello, I've found the Solidus platform that looks great, but I have never tried Ruby on Rails before. I think asking someone who has implemented this platform for ideas, recommendations, or solutions would be better than trying it myself.

My requirement is that I already have a mobile application, but I don't want to develop the user interface for it. Instead, I want to create a new tab for online shopping and use the Solidus platform. When a customer wants to purchase a product, they should be able to bypass the user profile and go directly to the order, or create a customer profile on the Solidus platform seamlessly. I'm not sure what the best way to do this with this platform is.

For example, when a user clicks on the shopping tab, the application could forward the user's token or identity to Solidus, which would then pull the customer's profile immediately after getting the user's identity. Or, when a user clicks on the shopping tab, the application could add the product to the cart and then, after the customer clicks on checkout and payment, Solidus could pull the user's identity.

**Comment by kennyadsl** [comment]
Hello @texiontech, sorry for the late reply. I am not able to assist you in shaping the architecture of the solution you want to build, and I have no experience with mobile application and their authentication best practices. Still, what you describe seems doable, and it's more a matter of understanding how it would work with a generic Rails application, Solidus is not much different!
